src/main/java/org/xerial/snappy/SnappyLoader.java

```diff
@@ -170,7 +177,10 @@ static synchronized SnappyApi loadSnappyApi()
         }
         catch(Exception e) {
             // Fall-back to pure-java Snappy implementation
-            setSnappyApi(new PureJavaSnappy());
+            if (order.equals(ByteOrder.LITTLE_ENDIAN))
+        		setSnappyApi(new PureJavaSnappy());
+        	else
+        		setSnappyApi(new PureJavaSnappyBE());
```

Comment:
This is a duplicated code. We can extract this part, for example, as `private static void loadPureJavaSnappy() { ... }`.

Reply:
I actually found that for AirCompressor snappy testing, the overhead of just doing the littleEndian conversion in specific places was insignificant.
Do we still want to have 2 sets of classes or just do the test in specific places in the RawCompress/RawDecompress methods ?